tokio/src/runtime/handle.rs

```diff
@@ -252,6 +252,15 @@ impl Handle {
     /// timers will not work unless there is another thread currently calling
     /// [`Runtime::block_on`] on the same runtime.
     ///
+    /// If this function is called from within a Tokio runtime's asynchronous
+    /// context, such as inside a [`Runtime::block_on`] scope, or from a function
+    /// annotated with the [`tokio::main`] attribute macro, it will panic, because
+    /// nesting runtimes is not allowed.
+    /// The same applies when calling `Handle::block_on` to re-enter the current
+    /// runtime. To safely run asynchronous code from such a context, you can use
+    /// [`task::block_in_place`] (only available with the multi-thread scheduler),
```

Comment:
Mentioning the `block_in_place` here might be too verbose, as there is already an example in below.